code/_core/obj/structure/interactive/lighting/fixture/_fixture.dm

```diff
@@ -37,10 +37,14 @@
 	else
 		create_destruction(T,list(/obj/item/material/sheet = 1),/material/steel)
 		. = ..()
+		if(src in SSgenerator_power.stationside_lights)
+			SSgenerator_power.stationside_lights -= src
```

Comment:
You don't need to check if src is in the stationside lights. It's safe, faster, and better to just remove it even though it might not exist in the first place (unless you're doing special behavior after it).